flutter_todo/lib/screens/homepage.dart

```diff
@@ -1,17 +1,21 @@
 import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
 import 'package:flutter_todo/components/todo_list.dart';
 import 'package:flutter_todo/components/create_item.dart';
 import 'package:flutter_todo/components/app_bar.dart';
-
+import 'package:flutter_todo/realm/realm_services.dart';
 class HomePage extends StatelessWidget {
   const HomePage({Key? key}) : super(key: key);
 
   @override
   Widget build(BuildContext context) {
-    return const Scaffold(
+    return Provider.of<RealmServices?>(context, listen: false) == null
+        ? Container()
```

Comment:
when would this be `null`?

and why is `listen: false`?

Reply:
If the user hasn't login to the app this `RealmServices` is not initialized (`create: (context) => null`), but the `HomePage` is already initialized because it is added to the `routes` in the `main.dart`.
With this code I'm preventing `TodoAppBar`, `TodoList` and `CreateItemAction` to be initialized, because they insist to have and instance of `RealmServices`. I return a `Container` instead of them.
`listen: false` is set to prevent `HomePage` to be reloaded every time when `notifyListeners` is called. We are rebuilding only specific widgets in that page, not the whole one.